cirq/value/digits_test.py

```diff
@@ -0,0 +1,61 @@
+import pytest
+
+import cirq
+
+
+def test_big_endian_bits_to_int():
+    assert cirq.big_endian_bits_to_int([0, 1]) == 1
+    assert cirq.big_endian_bits_to_int([1, 0]) == 2
+    assert cirq.big_endian_bits_to_int([0, 1, 0]) == 2
+    assert cirq.big_endian_bits_to_int([1, 0, 0, 1, 0]) == 18
+
+    assert cirq.big_endian_bits_to_int([]) == 0
+    assert cirq.big_endian_bits_to_int([0]) == 0
+    assert cirq.big_endian_bits_to_int([0, 0]) == 0
+    assert cirq.big_endian_bits_to_int([0, 0, 0]) == 0
+
+    assert cirq.big_endian_bits_to_int([1]) == 1
+    assert cirq.big_endian_bits_to_int([0, 1]) == 1
+    assert cirq.big_endian_bits_to_int([0, 0, 1]) == 1
+
+
+def test_big_endian_digits_to_int():
+    with pytest.raises(ValueError, match=r'len\(base\)'):
+        _ = cirq.big_endian_digits_to_int([1, 2, 3], base=[2, 3, 5, 7])
+    with pytest.raises(ValueError, match='Out of range'):
+        _ = cirq.big_endian_digits_to_int([105, 106, 107], base=4)
+
+    assert cirq.big_endian_digits_to_int([0, 1], base=102) == 1
+    assert cirq.big_endian_digits_to_int([1, 0], base=102) == 102
+    assert cirq.big_endian_digits_to_int([1, 0], base=[5, 7]) == 7
+    assert cirq.big_endian_digits_to_int([0, 1], base=[5, 7]) == 1
+    assert cirq.big_endian_digits_to_int([1, 2, 3, 4], base=[2, 3, 5, 7]) == 200
+    assert cirq.big_endian_digits_to_int([1, 2, 3, 4], base=10) == 1234
+
+
+def test_big_endian_int_to_bits():
+    assert cirq.big_endian_int_to_bits(2, bit_count=4) == [0, 0, 1, 0]
+    assert cirq.big_endian_int_to_bits(18,
+                                       bit_count=8) == [0, 0, 0, 1, 0, 0, 1, 0]
+    assert cirq.big_endian_int_to_bits(18, bit_count=4) == [0, 0, 1, 0]
+    assert cirq.big_endian_int_to_bits(-3, bit_count=4) == [1, 1, 0, 1]
+
+
+def test_big_endian_int_to_digits():
+    with pytest.raises(ValueError, match='Inconsistent digit count'):
+        _ = cirq.big_endian_int_to_digits(0, base=[], digit_count=1)
+    with pytest.raises(ValueError, match='Inconsistent digit count'):
+        _ = cirq.big_endian_int_to_digits(0, base=[2, 3], digit_count=20)
```

Comment:
Mypy doesn't complain about these?

Reply:
I don't think overloads have to be exhaustive. These meet the signature of the general method..

Reply:
It should.

https://mypy.readthedocs.io/en/latest/more_types.html#function-overloading
> For example, mypy will understand that mouse_event(5, 25) will always have a return type of ClickEvent and will **report errors for calls like mouse_event(5, 25, 2)**.

Reply:
Mypy accepts the following:

```python
@overload
def ff(x: int, y: None) -> bool:
    pass

@overload
def ff(x: None, y: int) -> str:
    pass

def ff(x: Optional[int], y: Optional[int]) -> Union[str, bool]:
    pass


def g():
    a = ff(1, None)
    b = ff(None, 1)
    c = ff(None, None)
    d = ff(1, 1)
```

So I'm inclined to think it's fine.